src/client/client_api/sequence_apis.rs

```diff
@@ -783,27 +783,18 @@ mod tests {
         let client = create_test_client().await?;
         let owner = client.public_key();
         let perms = BTreeMap::<PublicKey, SequencePrivatePermissions>::new();
-        let sequence_address = client
-            .store_private_sequence(None, name, tag, owner, perms)
-            .await?;
-
-        let mut balance_before_delete = client.get_balance().await?;
+        let sequence_address =
+            retry_loop!(client.store_private_sequence(None, name, tag, owner, perms.clone()));
 
-        while balance_before_delete == Token::from_str("0")?
-            || balance_before_delete == Token::from_str("10")?
-        {
-            sleep(Duration::from_millis(200)).await;
-            balance_before_delete = client.get_balance().await?;
-        }
+        let _ =
+            retry_loop_for_pattern!(client.get_balance(), Ok(bal) if *bal != Token::from_str("0")?);
+        let balance_before_delete = retry_loop_for_pattern!(client.get_balance(), Ok(bal) if *bal != Token::from_str("10")?)?;
```

Comment:
This is one more `get_balance()` than the impl before, I suppose `get_balance()` problably has no side effects, but could it change between the 2 calls?

Reply:
yeh, this is basically the same as previous, but the loop pattern matching doesnt let us (or I didn't figure out), how to get it to match against an or here.

Balance could change, but wont go to 0 during this test.

Reply:
```suggestion
                let balance_before_delete = retry_loop_for_pattern!(client.get_balance(), Ok(bal) if *bal != Token::from_str("10")? || *bal != Token::from_str("0")?)?;
```
Wouldn't this do the trick?